lnwire/features.go

```diff
@@ -315,6 +323,8 @@ var Features = map[FeatureBit]string{
 	ScidAliasOptional:                    "scid-alias",
 	ZeroConfRequired:                     "zero-conf",
 	ZeroConfOptional:                     "zero-conf",
+	RouteBlindingRequired:                "route-blinding",
```

Comment:
asking early here since the answer will I think affect the first 2 PRs:

1. should we add an option to _not_ advertise this bit?
2. Should we explicitly disallow nodes using us a blinded hop if we do not advertise the bit?

I know it will add a lot of checks in various places but the reason I ask is that while we are unsure of the usable pattern here, users might want to explicitly not want to support route blinding. 
The main reason I can think of for this is because, while we dont have PTLCs, route blinding opens up the door to easy or unintentional worm-holing: it can happen semi easily since a sender might choose a path to the intro node that actually includes a hop that is also included in the blinded route. The sender wont know that it is actually providing that node with the opportunity to worm-hole.

Reply:
ah, realised further along in the review of PR #2  we could potentially just check this once in `ValidateParsedPayloadTypes`....

Reply:
> should we add an option to not advertise this bit?

I don't think that we should go down the road of allowing disabling protocol features, seems like it'll cause us headaches in the future. We don't do this for things like data loss protect (which is much higher risk IMO, because you _effectively_ tell people you've lost your db even though _theoretically_ you could have a watchtower), so I don't think that we should do it for route blinding which are relatively much lower risk?

> Should we explicitly disallow nodes using us a blinded hop if we do not advertise the bit?

Not actually sure what the spec answer is here, but looking at features like `TLVOnionPayload` we'll forward TLV payloads even if we somehow flipped the feature off. 

Both of these questions seem like a higher level LND decision (/possibly even spec ponderance) so open to other opinions here, but my 2c would be to implement features the same way we've done in the past.

Reply:
> I don't think that we should go down the road of allowing disabling protocol features, seems like it'll cause us headaches in the future

We already allow nearly every protocol feature to be disabled, so it's pretty standard practice for us: https://github.com/lightningnetwork/lnd/blob/master/lncfg/protocol.go#L20-L56. 

IMO given this is still new code that exposes new code paths for a _forwarded_ HTLC (something any node on the network can trigger), we should definitely add a config flag to disable the feature. I don't think we can declaratively state this is lower risk than the other protocol features, as the feature still hasn't yet been widely deployed, therefore unknown unknowns lurk.

Given this code is pretty trivial, what exactly is the downside of _not_ adding a flag that lets users turn it off? If we look at the other extreme, I think the worst case is something like us introducing the equivalent of a ping of death (HTLC of death in this case) with no way of mitigating other than an emergency release or manually applied patch. 

> Should we explicitly disallow nodes using us a blinded hop if we do not advertise the bit?

Yes, based on the feature bit that translates into a switch/link config flag, we should treat the fields are unknown if someone attempts to utilize it but the feature if off.

Reply:
> We already allow nearly every protocol feature to be disabled, so it's pretty standard practice for us

My bad, I thought that the cfg options to turn things _off_ in here were mostly for itests - eg we don't have a `NoDataLossProtect` which is an option which I think it fair to say exposes risk?

Happy to add the flag, just thought it was pretty atypical.